### PR TITLE
clean up 705 and add rateQuanity with test

### PR DESCRIPTION
**Pull request by jgrant22**
add rateQuantity + test, cleaned  Dose and Rate yaml

### CODE REVIEW

File: src/main/java/io/github/linuxforhealth/hl7/data/SimpleDataValueResolver.java [hunk 1]
```diff
@@ -507,6 +507,20 @@
         return commonCodingSystemV2(table, code, text, version);
     };
 
+    public static final ValueExtractor<Object, String> UNIT_SYSTEM = (Object value) -> {
```
Comment (cragun47) [inline src/main/java/io/github/linuxforhealth/hl7/data/SimpleDataValueResolver.java:510]: As far as I can tell, there is no test of this code.  Not a specific unit test, nor any of the tests in `Hl7MedicationRequestFHIRConversionTest.java`.  I don't see any of them that have a system in x.3.
Please create at least one test that will exercise this code.

**Review by cragun47 — CHANGES_REQUESTED** [review]
Needs test which exercises UNIT_SYSTEM.
Otherwise mostly documentation clarifications.

File: src/main/resources/hl7/secondary/DoseAndRate.yml [hunk 2]
```diff
@@ -1,95 +1,62 @@
-# (C) Copyright IBM Corp. 2021
+# (C) Copyright IBM Corp. 2021, 2022
 #
 # SPDX-License-Identifier: Apache-2.0
 #
 ---
-rateRatio_1: # gets rateRatio from RXO
-  condition: $rxo21 NOT_NULL
+rateRatio: # gets rateRatio from RXO or RXE if RXO is not present
+  condition: $condition1 NOT_NULL && $condition2 NOT_NULL
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/main/resources/hl7/secondary/DoseAndRate.yml:7]: rateRatio looks at both RXO.21 and RXO.17. the condition separates it from rate Quantity.

File: src/main/resources/hl7/secondary/DoseAndRate.yml [hunk 3]
```diff
@@ -1,95 +1,62 @@
...
-    denunit: RXE.22
-    unitSystem: RXE.24.3 | $unitSys
-  constants:
-    unitSys: "http://unitsofmeasure.org"
-    denomVal: "1"
-
-doseQuantity_1: # gets doseQuantity from RXO
-  condition: $rxo2 NOT_NULL && $rxo3 NULL
+rateQuantity: # gets rateQuantity from from RXO or RXE if RXO is not present
+  condition: $condition1 NOT_NULL && $condition2 NULL
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/main/resources/hl7/secondary/DoseAndRate.yml:23]: fixed

File: src/main/resources/hl7/secondary/DoseAndRate.yml [hunk 4]
```diff
@@ -1,95 +1,62 @@
-# (C) Copyright IBM Corp. 2021
+# (C) Copyright IBM Corp. 2021, 2022
 #
 # SPDX-License-Identifier: Apache-2.0
 #
 ---
-rateRatio_1: # gets rateRatio from RXO
-  condition: $rxo21 NOT_NULL
+rateRatio: # gets rateRatio from RXO or RXE if RXO is not present
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/main/resources/hl7/secondary/DoseAndRate.yml:6]: fixed

File: src/main/java/io/github/linuxforhealth/hl7/data/SimpleDataValueResolver.java (hunk 1 above)
Comment (jgrant22) [reply src/main/java/io/github/linuxforhealth/hl7/data/SimpleDataValueResolver.java:510]: theres one on line 965 in medreqTestRateRation but I will use a different system

File: src/main/resources/hl7/secondary/Dosage.yml [hunk 5]
```diff
@@ -1,33 +1,16 @@
-# (C) Copyright IBM Corp. 2021
+# (C) Copyright IBM Corp. 2021, 2022
 #
 # SPDX-License-Identifier: Apache-2.0
 #
 ---
-maxDosePerPeriod_1: # gets max dose from RXO
-  condition: $rxo23 NOT_NULL
+maxDosePerPeriod_1: # gets max dose from RXO or RXE if RXO is not present
```
[the remark this answers is not in the document]
Comment (jgrant22) [reply src/main/resources/hl7/secondary/Dosage.yml]: fixed

**Review by cragun47 — APPROVED** [review]
Thanks for the changes, Jared.    
Approved with additional condition that you add .withValidateResource() on line 48 of ResourceUtils?  Somehow that got left off.

**Pull request merged**